Assets/Plugins/CountlySDK/Helpers/CountlyStorageHelper.cs

```diff
@@ -0,0 +1,125 @@
+
+        private CountlyLogHelper _logHelper;
+
+        internal SegmentDao ViewSegmentDao { get; private set; }
+        internal SegmentDao NonViewSegmentDao { get; private set; }
+
+        internal Dao<ConfigEntity> ConfigDao { get; private set; }
+        internal Dao<RequestEntity> RequestDao { get; private set; }
+        internal Dao<EventEntity> ViewEventDao { get; private set; }
+        internal Dao<EventEntity> NonViewEventDao { get; private set; }
+        internal Dao<EventNumberInSameSessionEntity> EventNrInSameSessionDao { get; private set; }
+
+        internal CountlyStorageHelper(CountlyLogHelper logHelper)
+        {
+            _logHelper = logHelper;
+
+            if (FirstLaunchAppHelper.IsFirstLaunchApp) {
+                _currentVersion = _schemaVersion;
+            } else {
+                _currentVersion = PlayerPrefs.GetInt(Constants.SchemaVersion, 0);
+            }
+        }
+
+        /// <summary>
+        /// Create database and tables
+        /// </summary>
+        private DB BuildDatabase(long dbNumber)
+        {
+            DB.Root(Application.persistentDataPath);
+            DB db = new DB(dbNumber);
+
+            db.GetConfig().EnsureTable<SegmentEntity>(EntityType.Configs.ToString(), "Id");
+            db.GetConfig().EnsureTable<RequestEntity>(EntityType.Requests.ToString(), "Id");
+            db.GetConfig().EnsureTable<EventEntity>(EntityType.ViewEvents.ToString(), "Id");
+            db.GetConfig().EnsureTable<EventEntity>(EntityType.NonViewEvents.ToString(), "Id");
+            db.GetConfig().EnsureTable<SegmentEntity>(EntityType.ViewEventSegments.ToString(), "Id");
+            db.GetConfig().EnsureTable<SegmentEntity>(EntityType.NonViewEventSegments.ToString(), "Id");
+
+            if (_currentVersion < _schemaVersion) {
+                db.GetConfig().EnsureTable<EventNumberInSameSessionEntity>(EntityType.EventNumberInSameSessions.ToString(), "Id");
+            }
+
+            return db;
+        }
+
+        /// <summary>
+        /// Open database connection and initialize data access objects.
+        /// </summary>
+        internal void OpenDB()
+        {
+            _logHelper.Debug("[CountlyStorageHelper] OpenDB");
+
+            _db = BuildDatabase(_dbNumber);
+            DB.AutoBox auto = _db.Open();
+
+            ViewSegmentDao = new SegmentDao(auto, EntityType.ViewEventSegments.ToString(), _logHelper);
+            NonViewSegmentDao = new SegmentDao(auto, EntityType.NonViewEventSegments.ToString(), _logHelper);
+
+            ConfigDao = new Dao<ConfigEntity>(auto, EntityType.Configs.ToString(), _logHelper);
+            RequestDao = new Dao<RequestEntity>(auto, EntityType.Requests.ToString(), _logHelper);
+            ViewEventDao = new Dao<EventEntity>(auto, EntityType.ViewEvents.ToString(), _logHelper);
+            NonViewEventDao = new Dao<EventEntity>(auto, EntityType.NonViewEvents.ToString(), _logHelper);
+
+            if (_currentVersion < _schemaVersion) {
```

Comment:
This is a bad "if" as this would be executed every time there is any migration happening. This needs to be done in only one migration.

Reply:
to move this to a place where migration happens, we need  'auto' reference there. 
we need to move 'auto' object to class level. (it will remain in memory till the app get kill).

**_This is a bad "if" as this would be executed every time there is any migration happening. This needs to be done in only one migration._**

this 'if' execute only one time but auto will remain in memory will destroy on app kill.


<img width="828" alt="Screenshot 2021-02-24 at 10 31 50 AM" src="https://user-images.githubusercontent.com/8924406/108953843-ac699900-768d-11eb-9e15-c0f4f10447bf.png">
,